packages/playwright/src/reporters/base.ts

```diff
@@ -489,11 +489,34 @@ export function stripAnsiEscapes(str: string): string {
   return str.replace(ansiRegex, '');
 }
 
+function characterWidth(c: string) {
+  return /[\u4e00-\u9fff]/.test(c) ? 2 : 1;
+}
+
+function stringWidth(v: string) {
+  let width = 0;
+  for (let i = 0; i < v.length; ++i)
+    width += characterWidth(v[i]);
```

Comment:
v[i] accesses a single 16-bit code unit, which may not represent a full character if it's part of a surrogate pair.

Reply:
good point. I updated it to segment with `Intl.Segmenter` in https://github.com/microsoft/playwright/pull/33322/commits/f778ddb6c50bb0ec32cfe7bef59c2f98fd9f12e5, that's working well with surrogate emoji in my testing.